scripts/code_generation/jsoncan/src/can_database.py

```diff
@@ -253,6 +266,20 @@ def node_alerts(self, node: str, alert_type: CanAlertType) -> List[str]:
             if node in self.alerts
             else []
         )
+        
+    def node_IDcodes(self, node: str, alert_type :CanAlert) -> Dict[str,int]:
+        
+        return(
+            {
+                alert:
+                codeID
+                for alert, codeID in self.alerts[node].items()
+                if alert.alert_type == alert_type
+            }
+            
+            if node in self.alerts
+            else {}
+        )
```

Comment:
nice! 🐍 

nit: our convention for python variables and functions is `snake_case`, can you change it here and in another files?
also, maybe pick a better name than `node_id_codes` and add some documentation to this function since `id_codes` is a bit ambiguous.

Reply:
ex. `node_alert_ids`